### PR TITLE
Various Optimizations

### PR DESCRIPTION
**Pull request by ghost**
Please refer to the commit messages for more information.

### CODE REVIEW

**Comment by con-f-use** [comment]
Hi again. thanks for https://github.com/instantOS/instantOS/discussions/126 and deciding to open this PR. It'll take a while until I'll be able to test the changes. Don't take it as ignoring your work, it's just  we've been burned before with compatibility issues of bash between systems. These kind of changes involving modern Bash features (not really so modern) need to be tested on a couple of arch-based and NixOS-based environments until they can be merged, because people are running instantOS tools on a surprising variety of systems. Also, not a lot is gained by these kind of optimizations but things can break, so as long as there is no proven bottleneck that is eliminated, these kind of changes are low priority and should also improve readability and maintainability of the code, while they are there. It's part of what Knuth meant by "premature optimizations...". Also we'd like to bundle as many of these more cosmetic changes into one testing run, to minimize the work. Talking about both this PR and the discussion in #126

**Comment by ghost** [comment]
No worries.

The use of `printf` in the first commit is about as standard as it gets, even working in Bourne Shell, if you were to use it on `"$@"` (since regular arrays aren't of course available). As for the 2nd commit, the use of `strftime()`-style printing with `printf` was apparently made available as of BASH 4.2, so, since InstantOS seems to be Arch-based, it should be fine. I completely understand wanting to test things, though.

I'm not sure when the `-v NAME` feature became available with `printf`, but I was able to find that it apparently doesn't work when assigning directly to an index in an array until BASH 4.1, which tells me the `-v` feature has been around for probably at least a bit longer than that.

I actually decided to just try `printf '%(%F)T\n' -1` in BASH 5.1, and it worked as expected. I assume that's what Arch uses, or whatever is the latest, if I'm out of the loop. The only thing left is to confirm all of the format specifications you need work, but it's pretty standard stuff, so I doubt it'll be an issue. According to the BASH man page, `printf` uses `strftime()` behind the scenes.

_However_! I have absolutely **zero** clue about NixOS. I contributed with Arch in mind, so I apologise if I missed something. I saw that InstantOS was Arch-based, so I didn't expect NixOS to be a variable. Lol

I hope that helps alleviate some concerns.

File: settings.sh [hunk 1]
```diff
@@ -80,7 +80,7 @@ searchall() {
         declare -p allsettings >"$CFG_CACHE" 2>/dev/null
     fi
     export SIDEBARINDEX=0
-    CHOICE=$(for k in "${!allsettings[@]}"; do echo "$k"; done | sidebar)
+    CHOICE=`printf '%s\n' "${!allsettings[@]}" | sidebar`
```
Comment (con-f-use) [inline settings.sh:83]: The backticks are of course an anachronism.

Comment (ghost) [reply settings.sh:83]: It's valid syntax, but I do ordinarily try to stick to syntax preferred by the project, so I apologise for straying from your preferred `$()` form of command substitution. I hadn't used the graves as part of the optimization, it was just habit.

**Comment by ghost** [comment]
One last thing, which might be of interest to someone on your end: Last night, I decided to just compile releases of BASH, sans _readline_ (I'm not too fussed with interactivity), between 3.0 and 5.1. I found that the basic use of `-v` with `printf` worked even as far back as BASH 3.1.

**Pull request merged**